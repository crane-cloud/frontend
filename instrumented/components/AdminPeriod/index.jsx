import React, { useState } from "react";
import PropTypes from "prop-types";
import "./AdminPeriod.css";

const AdminPeriod = (props) => {
  const [period, setPeriod] = useState("all");

  const handleChange = ({ target }) => {
    const { onChange } = props;
    setPeriod(target.getAttribute("value"));
    onChange(target.getAttribute("value"));
  };
  return (
    <div className="PeriodContainer">
      <div className="PeriodButtonsSection">
        <div
          className={`${period === "3m" && "PeriodButtonActive"} PeriodButton`}
          name="3month"
          value="3m"
          role="presentation"
          onClick={handleChange}
        >
          3m
        </div>
        <div
          className={`${period === "4m" && "PeriodButtonActive"} PeriodButton`}
          name="4months"
          value="4m"
          role="presentation"
          onClick={handleChange}
        >
          4m
        </div>
        <div
          className={`${period === "8m" && "PeriodButtonActive"} PeriodButton`}
          name="8months"
          value="8m"
          role="presentation"
          onClick={handleChange}
        >
          8m
        </div>
        <div
          className={`${period === "1y" && "PeriodButtonActive"} PeriodButton`}
          name="1year"
          value="1y"
          role="presentation"
          onClick={handleChange}
        >
          1y
        </div>
        <div
          className={`${period === "2y" && "PeriodButtonActive"} PeriodButton`}
          name="2year"
          value="2y"
          role="presentation"
          onClick={handleChange}
        >
          2y
        </div>
        <div
          className={`${period === "all" && "PeriodButtonActive"} PeriodButton`}
          name="all"
          value="all"
          role="presentation"
          onClick={handleChange}
        >
          all
        </div>
      </div>
    </div>
  );
};

AdminPeriod.propTypes = {
  onChange: PropTypes.func.isRequired,
};

export default AdminPeriod;
