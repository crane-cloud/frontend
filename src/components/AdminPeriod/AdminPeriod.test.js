import React from "react";
import { render } from "@testing-library/react";

import AdminPeriod from "./";

describe("Test the admin period component", () => {
  const onChange = jest.fn();
  const AdminPeriodComponent = render(<AdminPeriod onChange={onChange} />);
  it("checks if the admin period component matches the snapshot", () => {
    expect(AdminPeriodComponent).toMatchSnapshot();
  });
});
