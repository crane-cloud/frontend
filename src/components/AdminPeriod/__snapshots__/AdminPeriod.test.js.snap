// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the admin period component checks if the admin period component matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="PeriodContainer"
      >
        <div
          class="PeriodButtonsSection"
        >
          <div
            class="false PeriodButton"
            name="3month"
            role="presentation"
            value="3m"
          >
            3m
          </div>
          <div
            class="false PeriodButton"
            name="4months"
            role="presentation"
            value="4m"
          >
            4m
          </div>
          <div
            class="false PeriodButton"
            name="8months"
            role="presentation"
            value="8m"
          >
            8m
          </div>
          <div
            class="false PeriodButton"
            name="1year"
            role="presentation"
            value="1y"
          >
            1y
          </div>
          <div
            class="false PeriodButton"
            name="2year"
            role="presentation"
            value="2y"
          >
            2y
          </div>
          <div
            class="PeriodButtonActive PeriodButton"
            name="all"
            role="presentation"
            value="all"
          >
            all
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="PeriodContainer"
    >
      <div
        class="PeriodButtonsSection"
      >
        <div
          class="false PeriodButton"
          name="3month"
          role="presentation"
          value="3m"
        >
          3m
        </div>
        <div
          class="false PeriodButton"
          name="4months"
          role="presentation"
          value="4m"
        >
          4m
        </div>
        <div
          class="false PeriodButton"
          name="8months"
          role="presentation"
          value="8m"
        >
          8m
        </div>
        <div
          class="false PeriodButton"
          name="1year"
          role="presentation"
          value="1y"
        >
          1y
        </div>
        <div
          class="false PeriodButton"
          name="2year"
          role="presentation"
          value="2y"
        >
          2y
        </div>
        <div
          class="PeriodButtonActive PeriodButton"
          name="all"
          role="presentation"
          value="all"
        >
          all
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
